### PR TITLE
Make well-known IPv4 addresses readonly

### PR DESCRIPTION
**Pull request by nil4**
Fix https://github.com/dotnet/corefx/issues/33373 using the approach suggested in PR https://github.com/dotnet/corefx/pull/33476 

/cc @alfredmyers @stephentoub

### CODE REVIEW

File: src/System.Net.Primitives/tests/FunctionalTests/IPAddressTest.cs [hunk 1]
```diff
@@ -311,6 +311,15 @@ public static void Address_Property_Success()
             Assert.Equal(ip1, ip2);
             Assert.Equal(ip1.GetHashCode(), ip2.GetHashCode());
         }
+
+        [Fact]
```
Comment (stephentoub) [inline src/System.Net.Primitives/tests/FunctionalTests/IPAddressTest.cs]: This is going to fail when it runs on .NET Framework.  Search the repo for the SkipOnTargetFramework attribute usage, and apply a similar attribute here to skip it when running on netfx.

Comment (nil4) [reply src/System.Net.Primitives/tests/FunctionalTests/IPAddressTest.cs]: @stephentoub addressed in latest commit

File: src/System.Net.Primitives/src/System/Net/IPAddress.cs [hunk 2]
```diff
@@ -532,6 +532,10 @@ public long Address
                 {
                     if (PrivateAddress != value)
                     {
+                        if (this is ReadOnlyIPAddress)
+                        {
+                            throw new SocketException(SocketError.OperationNotSupported);
```
Comment (alfredmyers) [inline src/System.Net.Primitives/src/System/Net/IPAddress.cs]: @nil4 , @stephentoub 
Is `SocketException` the best option here? Given the example at https://github.com/dotnet/corefx/issues/33373#issuecomment-438515269 it seems a little out of context.

Comment (stephentoub) [reply src/System.Net.Primitives/src/System/Net/IPAddress.cs]: It's what's already thrown when you try to set the address of an IPv6 IPAddress, and it's the only exception type documented for IPAddress.Address to throw.

File: src/System.Net.Primitives/tests/FunctionalTests/IPAddressTest.cs (hunk 1 above)
Comment (alfredmyers) [reply src/System.Net.Primitives/tests/FunctionalTests/IPAddressTest.cs]: @stephentoub Any chance of this patch eventually making it's way into the .NET Framework?

File: src/System.Net.Primitives/src/System/Net/IPAddress.cs (hunk 2 above)
Comment (nil4) [reply src/System.Net.Primitives/src/System/Net/IPAddress.cs]: @alfredmyers I kept it consistent with the error thrown above 

https://github.com/dotnet/corefx/blob/ba6a1037c4e3dd0f0cc46a02a8a2b649e6adec7c/src/System.Net.Primitives/src/System/Net/IPAddress.cs#L539-L542

where changing an IPv6 `Address` value is blocked. My assumption was that this exception would already be documented, and did not want to introduce another type (and error message string) for blocking the same thing here.

File: src/System.Net.Primitives/tests/FunctionalTests/IPAddressTest.cs (hunk 1 above)
Comment (stephentoub) [reply src/System.Net.Primitives/tests/FunctionalTests/IPAddressTest.cs]: > Any chance of this patch eventually making it's way into the .NET Framework?

Very doubtful.

File: src/System.Net.Primitives/src/System/Net/IPAddress.cs (hunk 2 above)
Comment (alfredmyers) [reply src/System.Net.Primitives/src/System/Net/IPAddress.cs]: LGTM

**Comment by davidsh** [comment]
@dotnet-bot test Outerloop Windows x64 Debug Build
@dotnet-bot test Outerloop Linux x64 Debug Build
@dotnet-bot test Outerloop NETFX x86 Debug Build
@dotnet-bot test Outerloop UWP CoreCLR x64 Debug Build
@dotnet-bot test Outerloop OSX x64 Debug Build

**Comment by nil4** [comment]
As far as I can see, the build failures do not seem related to this change. Happy to address any remaining issues if I'm wrong about this.

**Comment by ahsonkhan** [comment]
@dotnet-bot test Outerloop Windows x64 Debug Build
@dotnet-bot test Outerloop Linux x64 Debug Build
@dotnet-bot test Outerloop NETFX x86 Debug Build
@dotnet-bot test Outerloop UWP CoreCLR x64 Debug Build

**Comment by ahsonkhan** [comment]
@dotnet-bot test UWP CoreCLR x64 Debug Build

**Comment by stephentoub** [comment]
None of these failures are associated with this change.

Thanks for the fix.

**Pull request merged**